lib/sequelize.js

```diff
@@ -517,41 +525,47 @@ class Sequelize {
       }
 
       if (options.transaction && options.transaction.finished) {
-        const error = new Error(options.transaction.finished+' has been called on this transaction('+options.transaction.id+'), you can no longer use it. (The rejected query is attached as the \'sql\' property of this error)');
+        const error = new Error(`${options.transaction.finished} has been called on this transaction(${options.transaction.id}), you can no longer use it. (The rejected query is attached as the \'sql\' property of this error)`);
         error.sql = sql;
         return Promise.reject(error);
       }
 
-      if (this.test._trackRunningQueries) {
+      if (isFirstTry && this.test._trackRunningQueries) {
```

Comment:
Not related to this PR but a possible fix for randomly failing tests due to tracking queries //cc @janmeier

Reply:
Could very well be!

Reply:
that would be awesome!